cmd/gateway-azure.go

```diff
@@ -123,10 +125,22 @@ func azureToObjectError(err error, params ...string) error {
 }
 
 // Inits azure blob storage client and returns AzureObjects.
-func newAzureLayer(endPoint string, account, key string, secure bool) (GatewayLayer, error) {
+func newAzureLayer(args []string) (GatewayLayer, error) {
+	endPoint, secure, err := parseGatewayEndpoint(args[0])
```

Comment:
If each gateway will take care of its arguments, you need to check if `len(args) > 0` here